awx/main/models/inventory.py

```diff
@@ -820,9 +820,47 @@ def _get_related_jobs(self):
 
 
 class HostMetric(models.Model):
-    hostname = models.CharField(primary_key=True, max_length=512)
+    hostname = models.CharField(unique=True, max_length=512)
     first_automation = models.DateTimeField(auto_now_add=True, null=False, db_index=True, help_text=_('When the host was first automated against'))
     last_automation = models.DateTimeField(db_index=True, help_text=_('When the host was last automated against'))
+    last_deleted = models.DateTimeField(null=True, db_index=True, help_text=_('When the host was last deleted'))
+    automated_counter = models.BigIntegerField(default=0, help_text=_('How many times was the host automated'))
+    deleted_counter = models.BigIntegerField(default=0, help_text=_('How many times was the host deleted'))
+    deleted = models.BooleanField(
+        default=False, help_text=_('Boolean flag saying whether the host is deleted and therefore not counted into the subscription consumption')
+    )
+    used_in_inventories = models.BigIntegerField(null=True, help_text=_('How many inventories contain this host'))
```

Comment:
I recognize that it's hard to draw an exact line between integer vs. big integer fields. But I think a host (even localhost) will not be used in over 2 billion inventories.

Reply:
it sounds reasonable, maybe the counters don't need Big int as well
cc @Ladas

Reply:
@AlanCoding @slemrmartin yeah that is fair. You're right that the counter will not likely hit 2B of automations/deletes. So for the safe of storage saving, we can make them just Int,